scripts/leetcode.js

```diff
@@ -520,12 +538,65 @@ const loader = setInterval(() => {
           problemName + language,
           probStats,
           'upload',
+          // callback is called when the code upload to git is a success
+          () => { 
+            if(uploadState['countdown']) clearTimeout(uploadState['countdown']); 
+            delete uploadState['countdown']
+            uploadState.uploading = false; 
+            markUploaded(); 
+          }
         ); // Encode `code` to base64
       }, 1000);
     }
   }
 }, 1000);
 
+/* Since we dont yet have callbacks/promises that helps to find out if things went bad */
+/* we will start 10 seconds counter and even after that upload is not complete, then we conclude its failed */
+function startUploadCountDown() {
```

Comment:
i like the overall idea, but 10seconds is a lot. Currently, we push the code when all test cases are passed. To check for that, we simply check for "Success" in the top left corner of the page. 
What if u instead create an event listener whenever we pressed the submit button and look for [success tag](https://github.com/QasimWani/LeetHub/blob/main/scripts/leetcode.js#L397). Once that condition is true, and whle it's submitting u can keep the spinner running.
But the idea is that as soon as someone hits submit, the spinner is activated.

Reply:
The spinner is activated by the success tag. The spinner will shut down and show a green check when code upload is successful. The 10s countdown is just for the case when the upload is unsuccessful. It's just if the code is not uploaded within 10s, then there probably was an error. But if there are no errors, spinners go away like in 3/4 seconds when the upload is complete. Does this make sense?

Reply:
10s thing is just an edge case handler when pushing to the Github repo fails. For example, if there are any issues with the OAuth tokens. Please let me know if that makes sense.